### DISCUSSION TITLE
Deletion of temp files

### DISCUSSION THREAD

**Discussion by RAmbika** [ImageMagick/ImageMagick#2145] [General]
Hi,
when imagemagick does fail(with timeout) to convert a document, temp files are left in the temp folder.
We are trying to clean up temp files with the following Approach:
A solution to create a temporary folder for each call to ImageMagick and point temp path to ImageMagick. 
That way when the process is complete, simply remove that folder to ensure temporary files are removed without interfering with other ImageMagick processes.

We found in your forums assign temp path to Enviroment variables like MAGICK_TEMPORARY_PATH, Magick_TMPDIR. If we use environment variables , these are applying for all conversions.
We need to assign temp path for each conversion. Please let us know , shall we assign temp path in this function 'SetImageRegistry' instead of using environment variables.

Thanks,
Ambika

**Comment by urban-warrior** [comment]
ImageMagick should delete all temporary files other than exiting due to signals it cannot catch.  If you provide us with instructions for reproducing the problem, we'll investigate.  

For our online ImageMagick dashboard @ https://imagemagick.org/MagickStudio/scripts/MagickStudio.cgi, we check the dates of any left-behind temporary files and expire them if they are more then 8 hours old.

On a per-process basis, you can set the temporary path from the command line with `-define registry:temporary-path=/magick/tmp` or from the API, as you suggest, SetImageRegistry().

**Comment by RAmbika** [comment]
Hi,

We set environment variable MAGICK_TEMPORARY_PATH=/magick/tmp.
We have some left over temp files in the temp folder.
We are making setImageRegistry() call before starting of every conversion. We have left-behind temporary files and expire them if they are more then 8 hours old, but files are not still deleted.
Please let us know if we missed anything?

MagickBooleanType result = SetImageRegistry(MagickCore::RegistryType::StringRegistryType, "temporary-path", "/magick/tmp", exception);
